examples/connext_dds/dynamic_data_request_reply/cpp/PrimeNumberReplier.cxx

```diff
@@ -0,0 +1,328 @@
+
+#include <math.h>
+#include <iostream>
+#include <vector>
+#include "PrimesType.cxx"
+
+
+
+
+/*
+ * Include the request-reply API
+ */
+#include "ndds/ndds_requestreply_cpp.h"
+
+class PrimeNumberReplierExample {
+
+private:
+    DDS::DomainParticipant *participant;
+
+public:
+
+    PrimeNumberReplierExample(int domain_id)
+    {
+        /* Create the participant */
+        participant = DDS::DomainParticipantFactory::get_instance()->
+                create_participant(
+                        domain_id,
+                        DDS::PARTICIPANT_QOS_DEFAULT,
+                        NULL /* listener */,
+                        DDS::STATUS_MASK_NONE);
+
+        if (participant == NULL) {
+            throw std::runtime_error("create_participant error");
+        }
+    }
+
+    ~PrimeNumberReplierExample()
+    {
+        /* Delete the participant and all contained entities */
+
+        DDS::ReturnCode_t retcode = participant->delete_contained_entities();
+        if (retcode != DDS::RETCODE_OK) {
+            fprintf(stderr, "delete_contained_entities error %d\n", retcode);
+        }
+
+        retcode = DDS::DomainParticipantFactory::get_instance()->
+            delete_participant(participant);
+        if (retcode != DDS::RETCODE_OK) {
+            fprintf(stderr, "delete_participant error %d\n", retcode);
+        }
+    }
+
+private:
+
+    void calculate_and_send_primes(
+        connext::Replier<DDS_DynamicData, DDS_DynamicData>& replier,
+        const DDS::SampleIdentity_t& request_sample_identity,
+        DDS_Long n,
+        DDS_Long primes_per_reply,
+        DDS_DynamicData *reply)
+        {
+
+        /* Set Status to REPLY_IN_PROGRESS */
+        reply->set_long("status", DDS_DYNAMIC_DATA_MEMBER_ID_UNSPECIFIED,0);
+
+        /* Set the sequence: */
+        DDS_LongSeq primes_seq;
+        primes_seq.maximum(primes_per_reply);
+
+        /* prime[i] indicates if i is a prime number
+         * Initially, we assume all of them are prime
+         */
+        std::vector<bool> prime(n+1, true);
```

Comment:
why do we need n+1?

Reply:
It could be the case where "n" is also a prime number (e.g: n=113). Hence, starting from 0 to "n" we need to allocate "n+1" memory space.